nix-buffer.el

```diff
@@ -152,42 +157,56 @@ EVENT The process status change event string."
 	  (kill-buffer out-buf)
 	  (kill-buffer err-buf))))))
 
-(defun nix-buffer--nix-build (root expr-file)
+(defun nix-buffer--nix-build (expr-file &optional root skip-safety)
   "Start the nix build.
+EXPR-FILE The file containing the nix expression to build.
+
 ROOT The path we started from.
 
-EXPR-FILE The file containing the nix expression to build."
+SKIP-SAFETY whether to skip the safety checks."
   (let* ((state-dir (f-join nix-buffer-directory-name
-			    (nix-buffer--unique-filename root)))
+			    (nix-buffer--unique-filename (or root
+							     buffer-file-name))))
 	 (out-link (f-join state-dir "result"))
 	 (current-out (file-symlink-p out-link))
-	 (err (generate-new-buffer " nix-buffer-nix-build-stderr")))
+	 (err (generate-new-buffer " nix-buffer-nix-build-stderr"))
+	 (command (list "nix-build" expr-file
+			"--out-link" out-link)))
     (ignore-errors (make-directory state-dir t))
+    (when root
+      (push "--arg" 'command)
+      (push "root" 'command)
+      (push root 'command))
     (make-process
      :name "nix-buffer-nix-build"
      :buffer (generate-new-buffer " nix-buffer-nix-build-stdout")
-     :command (list
-	       "nix-build"
-	       "--arg" "root" root
-	       "--out-link" out-link
-	       expr-file
-	       )
+     :command command
      :noquery t
      :sentinel (apply-partially 'nix-buffer--sentinel
 				out-link
 				current-out
 				expr-file
 				(current-buffer)
-				err)
+				err
+				skip-safety)
      :stderr err)
     (when current-out
-      (nix-buffer--load-result expr-file current-out))))
+      (nix-buffer--load-result expr-file current-out skip-safety))))
 
 (defcustom nix-buffer-root-file "dir-locals.nix"
   "File name to use for determining Nix expression to use."
   :group 'nix-buffer
   :type '(string))
 
+;;;###autoload
+(defun nix-buffer-with-string (expression)
+  "Start ‘nix-buffer’ but with a string EXPRESSION."
+  (interactive)
+  (let ((expr-file (make-temp-file "nix-buffer")))
```

Comment:
Hm, can't we just use `nix-build -E` instead of making a temp file?

Reply:
Actually that's probably a good idea. I had thought it would save on code duplication but it won't make a big difference.